x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -203,6 +205,14 @@ static boolean checkSameUserPermissions(String action, TransportRequest request,
                     return false;
                 }
                 final String username = usernames[0];
+                // Cross cluster access user can perform has privilege check
+                if (authentication.isCrossClusterAccess() && HasPrivilegesAction.NAME.equals(action)) {
+                    assert request instanceof HasPrivilegesRequest;
+                    return ((Authentication) authentication.getAuthenticatingSubject()
```

Comment:
I considered that. But I was a bit hesitate to add such method to `Authentication` because we don't even have a `getMetadata` method for it since `metadata` right now is a subject level information. So for now I added `getAuthenticationFromCrossClusterAccessMetadata` as a static method to the Authentication class. We can iterate in future to further rationalize how Authentication information should be distributed.